ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/ethernet_interfaces.py

```diff
@@ -95,6 +95,11 @@ def ethernet_interfaces(self) -> dict | None:
                                         interface["ospf_authentication"] = ospf_authentication
                                         interface["ospf_message_digest_keys"] = ospf_keys
 
+                            vrf_l3_multicast_enabled = default(get(vrf, "l3_multicast.enabled"), get(tenant, "l3_multicast.enabled"))
+
+                            if pim_config := get(l3_interface, "pim.enabled") and vrf_l3_multicast_enabled:
```

Comment:
https://peps.python.org/pep-0572/#relative-precedence-of

given that `and` has better precedence than `:=` I would recommend changing this to

```suggestion
                            if (pim_config := get(l3_interface, "pim.enabled")) and vrf_l3_multicast_enabled:
```

as it may not work correctly

Reply:
Ah, good catch.  Still getting used to this inline assignment nonsense :)